### DISCUSSION TITLE
How do I nest a signed JWT in an encrypted JWT?

### DISCUSSION THREAD

**Discussion by Evertt** [panva/jose#272] [Q&A] [answered]
I swear I first used the search function, but the threads that came up didn't seem to be about my question specifically.

I think I _almost_ understand how to do it.. I'm just missing one or two essential bits of information. In [this article](https://connect2id.com/products/nimbus-jose-jwt/examples/signed-and-encrypted-jwt), which is about a Java version of Jose, they explain that when you're nesting a signed JWT into an encrypted JWT, you need to specify during the building of the encrypted JWT that the content type is a JWT.

I don't see a method documented [here](https://github.com/panva/jose/blob/main/docs/classes/jwt_encrypt.EncryptJWT.md#readme) that allows me to do that.

I believe I _could_ also skip that step completely and just encrypt the signed JWT as if it were an arbitrary string. But I'd like to do things the canonical way if I can. So is there a more canonical way of nesting a signed JWT inside an encrypted JWT with jose?

**Comment by panva** [comment, marked as answer]
> I don't see a method documented here that allows me to do that.

```js
.setProtectedHeader({ cty: 'JWT', ...rest })
```

An example, without setting the content-type is in #112

**Reply by Evertt** [reply]
Thanks, that solves that question. Can I ask one last one? Why does `generateKeyPair()` only generate a key pair that can be used for signing? Is there also a way to generate a key pair for encrypting?

**Reply by panva** [reply]
What makes you think `generateKeyPair()` only generates a key pair that can be used for signing?

**Reply by Evertt** [reply]
Sorry, I was using the wrong algorithm I now realize.
